app/controllers/api/application_controller.rb

```diff
@@ -11,17 +15,31 @@ class Api::ApplicationController < ActionController::API
 
   after_action :verify_authorized, except: %i[index], unless: :skip_pundit?
   after_action :verify_policy_scoped, only: %i[index], unless: :skip_pundit?
+  after_action :set_csrf_cookie
 
   def index
     # This parent application controller throws errors from above after_actions
     # if the subclass does not implement this method. This is provided as a fallback.
     raise NotImplementedError, "The index method is not implemented."
   end
 
+  private
+
+  def set_csrf_cookie
+    cookies["CSRF-TOKEN"] = {
```

Comment:
I'm a little confused by this.
I thought the whole point of CSRF protection was to not rely on cookies?
Isn't the idea that the cookie is automatically associated and attached by the browser based on the destination of the request, not the origin of the request? Is this cookie set every time the HTTP only auth is set?

Reply:
See our conversation here for more details: https://laterolabs.slack.com/archives/CP688FUUQ/p1724361988634719